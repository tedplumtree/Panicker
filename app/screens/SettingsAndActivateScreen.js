import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { Card, Subheader } from 'react-native-material-ui';
import ToolbarCollapsingScrollLayout from '../layouts/ToolbarCollapsingScrollLayout';

const styles = StyleSheet.create({
  cardText: {
    padding: 15,
    paddingTop: 0,
  },
});

const SettingsAndActivateScreen = () => (
  <ToolbarCollapsingScrollLayout toolbar={{ centerElement: 'Panicker' }}>
    <Card>
      <Subheader text="Welcome!" />
      <Text style={styles.cardText}>
         This app will scream if someone tries to steal your phone. Click
         the action button below to drop the pin at your location. The best
         part about it is that you don&apos;t need to enable location tracking!
      </Text>
    </Card>
    {/* TODO: Card with settings */}
    {/* TODO: Floating action button */}
  </ToolbarCollapsingScrollLayout>
);

export default SettingsAndActivateScreen;
