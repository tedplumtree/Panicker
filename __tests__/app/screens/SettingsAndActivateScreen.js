import 'react-native';
import React from 'react';
import { ThemeProvider } from 'react-native-material-ui';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import SettingsAndActivateScreen from '../../../app/screens/SettingsAndActivateScreen';

const dummyMaterialDesignTheme = {
  palette: {
    primaryColor: '#009688',
  },
  toolbar: {
    container: {
      height: 123,
    },
  },
};

jest.mock(
  '../../../app/layouts/ToolbarCollapsingScrollLayout',
  () => 'ToolbarCollapsingScrollLayout',
);

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider uiTheme={dummyMaterialDesignTheme}>
      <SettingsAndActivateScreen />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
