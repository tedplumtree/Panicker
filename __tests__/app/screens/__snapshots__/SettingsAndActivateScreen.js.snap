// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ToolbarCollapsingScrollLayout
  toolbar={
    Object {
      "centerElement": "Panicker",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "marginHorizontal": 8,
          "marginVertical": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
          "zIndex": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 48,
            "justifyContent": "center",
          },
          Object {
            "paddingLeft": 16,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 24,
            },
            undefined,
          ]
        }
      >
        Welcome!
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "padding": 15,
          "paddingTop": 0,
        }
      }
    >
      This app will scream if someone tries to steal your phone. Click the action button below to drop the pin at your location. The best part about it is that you don't need to enable location tracking!
    </Text>
  </View>
</ToolbarCollapsingScrollLayout>
`;
